src/components/stake/StakeModal.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+    <!-- Pass down all attributes not declared as props -->
+    <Modal v-bind="$attrs" v-on="$listeners" class="stake-modal">
+        <template v-if="page === 1">
+            <StakeInfoPage @next="page += 1"/>
+        </template>
+        <template v-if="page === 2">
+            <StakeValidatorPage @back="page -= 1" @next="page += 1"/>
+        </template>
```

Comment:
If the template only contains one child node, you can omit the template and apply the `v-if` directly on the child.